.github/workflows/ci-master.yml

```diff
@@ -29,136 +29,49 @@ jobs:
     needs: Lint
     strategy:
       matrix:
-        os: [macos-latest, ubuntu-latest]
-        python-version: [3.6, 3.7, 3.8, 3.9]
-        exclude:
-          # conserve some osx builders
-          - python-version: 3.6
-            os: macos-latest
-          # conserve some osx builders
-          - python-version: 3.9
-            os: macos-latest
-        include:
-          - python-version: 3.6
-            os: ubuntu-latest
-            tox-target: py36
-          - python-version: 3.7
-            os: ubuntu-latest
-            tox-target: py37
-          - python-version: 3.8
-            os: ubuntu-latest
-            tox-target: py38-cov
-          - python-version: 3.9
-            os: ubuntu-latest
-            tox-target: py39
-          - python-version: 3.7
-            tox-target: py37
-            os: macos-latest
-          - python-version: 3.8
-            tox-target: py38
-            os: macos-latest
+        os: [ubuntu-latest, macos-latest, windows-latest]
+        python-version: [3.9]
     steps:
       - uses: actions/checkout@v2
         with:
           fetch-depth: 2
+
       - name: Set up Python ${{ matrix.python-version }}
         uses: actions/setup-python@v2
         with:
           python-version: ${{ matrix.python-version }}
-      - name: Set up cache
-        uses: actions/cache@v1
-        with:
-          path: .tox/${{ matrix.tox-target }}
-          key: venv-${{ runner.os }}-${{ matrix.python-version }}-${{ hashFiles('**/requirement*.txt') }}
-      - name: Install Test Runner
-        run: |
-          python -m pip install --upgrade pip
-          pip install tox
-      - name: Install Deps
-        run: |
-          tox -e ${{ matrix.tox-target }} --notest
-      - name: Test
-        run: |
-          tox -vv -e ${{ matrix.tox-target }}
-
-      - name: Coverage Generate XML
-        if: contains(matrix.tox-target, 'py38-cov')
-        run: |
-          ./.tox/${{ matrix.tox-target }}/bin/coverage xml && \
-          bash <(curl -s https://codecov.io/bash) -Z \
-          -v -f coverage.xml
 
-      - name: License Check
-        if: contains(matrix.tox-target, 'py39')
+      - name: Bootstrap poetry
+        shell: bash
         run: |
-          ./.tox/${{ matrix.tox-target }}/bin/python tools/dev/license-check.py
+          python -m pip install --upgrade poetry==1.1.8
 
-  Test-Windows:
-    # windows can't use the fast cache technique we use in our matrix builds
-    # where we cache the entire tox virtualenv directory, without error.
-    # instead we just cache the pip downloads, and install every time.
+      - name: Configure poetry
+        shell: bash
+        run: poetry config virtualenvs.in-project true
```

Comment:
This is creating a separate venv for each package because [the `make install-poetry` changes directories during installation](https://github.com/cloud-custodian/cloud-custodian/pull/6883/files#diff-76ed074a9305c04054cdebb9e9aad2d818052b07091de1f20cad0bbac34ffb52R11): 

```
Run make install-poetry
poetry install
Creating virtualenv c7n in /home/runner/work/cloud-custodian/cloud-custodian/.venv
Installing dependencies from lock file

Package operations: 68 installs, 0 updates, 0 removals

  • Installing pycparser (2.20)
...
  • Installing vcrpy (4.1.1)

Installing the current project: c7n (0.9.14)
for pkg in tools/c7n_gcp tools/c7n_azure tools/c7n_kube tools/c7n_openstack tools/c7n_mailer tools/c7n_logexporter tools/c7n_policystream tools/c7n_trailcreator tools/c7n_org tools/c7n_sphinxext tools/c7n_terraform; do cd $pkg && poetry install && cd ../..; done
Creating virtualenv c7n-gcp in /home/runner/work/cloud-custodian/cloud-custodian/tools/c7n_gcp/.venv
...
```

https://github.com/cloud-custodian/cloud-custodian/pull/6883/checks?check_run_id=3517237832#step:8:84

I guess this is the cause for the following failure:

```
==================================== ERRORS ====================================
________________________ ERROR collecting test session _________________________
Traceback (most recent call last):
  File "/opt/hostedtoolcache/Python/3.9.6/x64/lib/python3.9/importlib/__init__.py", line 127, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 1030, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1007, in _find_and_load
  File "<frozen importlib._bootstrap>", line 986, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 680, in _load_unlocked
  File "/home/runner/work/cloud-custodian/cloud-custodian/.venv/lib/python3.9/site-packages/_pytest/assertion/rewrite.py", line 170, in exec_module
    exec(co, module.__dict__)
  File "/home/runner/work/cloud-custodian/cloud-custodian/tools/c7n_gcp/tests/conftest.py", line 9, in <module>
    from c7n_gcp.client import get_default_project
ModuleNotFoundError: No module named 'c7n_gcp'
```

https://github.com/cloud-custodian/cloud-custodian/pull/6883/checks?check_run_id=3517237832#step:9:25

Reply:
Aha, thanks!!